command/agent/acl_endpoint.go

```diff
@@ -138,8 +138,11 @@ func (s *HTTPServer) ACLTokenBootstrap(resp http.ResponseWriter, req *http.Reque
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 
+	secret := req.Header.Get("X-Nomad-Bootstrap-Token")
```

Comment:
Why are we using a header instead of the existing `Secret` (or `BootstrapSecret` as I suggest elsewhere) field? I don't think there's anywhere else we *set* data using a header. :thinking:

Reply:
I was using the CSI driver as an example when I did the initial experiment.
The reason for this was an easy way to get the token passed from the client to the API that least impacted any other code.

Reply:
@schmichael do you want me to change anything here?

Reply:
Yes, let's switch to putting the bootstrap token into the request body to follow what most of our other APIs do. Since `ACLTokenBootstrapRequest.BootstrapSecret` already exists you can use it in the CLI and API.

Force-rescheduling a job is a similar example to follow of adding a request body to a command that didn't have one before:
CLI: https://github.com/hashicorp/nomad/blob/v1.3.0/command/job_eval.go#L116-L119
API: https://github.com/hashicorp/nomad/blob/v1.3.0/command/agent/job_endpoint.go#L110-L126

Reply:
Thanks I will check it out.

Reply:
Done